emcc.py

```diff
@@ -1131,6 +1161,10 @@ def add_link_flag(i, f):
       elif arg == '-':
         input_files.append((i, arg))
         newargs[i] = ''
+
+    if not input_files and not link_flags:
```

Comment:
why are link flags checked here? `no input files` seems like it should be shown regardless of the link flags?

Reply:
I'm not changing that, its just preexisting behaviour to match gcc/clang.   

Running clang with no linker flags generates `no input files`:

```
$ clang -Dxx
clang: error: no input files
````

Running clang with one or more linker flags will call the linker:

```
$ clang -Dxx -Wl,--pie
/usr/bin/ld: /usr/bin/../lib/gcc/x86_64-linux-gnu/10/crtbegin.o: relocation R_X86_64_32 against hidden symbol `__TMC_END__' can not be used when making a PIE object
/usr/bin/ld: /usr/bin/../lib/gcc/x86_64-linux-gnu/10/../../../x86_64-linux-gnu/crt1.o: in function `_start':
(.text+0x20): undefined reference to `main'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I'm just moving the code up so it runs as early as it can.